### PR TITLE
product page - gallery zoom photoswipe

### LINKED ISSUE #421: Add a "photoswipe" gallery behavior on product page images in order to be able to zoom

### PR DESCRIPTION
**Pull request by JBLach**
| Questions         | Answers
| ------------------| -------------------------------------------------------
| Description?      | If I understood the task correctly, I added a slider for the photo gallery in the modal, after clicking on the icon, we should see the correct photo and the modal with the slider, I also found a bug when we added the `display flex`  to the class active in slider, it looks wrong. below is an example
| Type?             | bug fix / improvement
| BC breaks?        |  no
| Deprecations?     |  no
| Fixed ticket?     | Fixes #421 

## Active class with display: flex;
https://user-images.githubusercontent.com/48727835/213734964-de8a7ad4-acc4-403e-b68f-1d523a35dc29.mp4

## Active class without display: flex;
https://user-images.githubusercontent.com/48727835/213735168-b9ea5460-2fc2-41c5-9ea5-361dff4b9ef4.mp4

## Appearance after changes
https://user-images.githubusercontent.com/48727835/213735317-54c90b04-98dc-4865-825c-d0d27566fc1f.mp4

## Mobile view
mobile_photo

### CODE REVIEW

**Comment by Hlavtox** [comment]
@JBLach The idea is to implement photoswipe fullscreen gallery, with zoom capabilities etc.

Check my implementation here - https://eshop-blanco.cz/drezy-s-odkapem/17943-kuchynsky-drez-blanco-faron-xl-6-s-antracit-4020684703345.html

**Comment by JBLach** [comment]
> @JBLach The idea is to implement photoswipe fullscreen gallery, with zoom capabilities etc.
> 
> Check my implementation here - https://eshop-blanco.cz/drezy-s-odkapem/17943-kuchynsky-drez-blanco-faron-xl-6-s-antracit-4020684703345.html
Hi @Hlavtox,
Is the photoswipe library neccessary here? I think the only thing it adds to us is the zoom option. I don't think this is the optimal solution, unless there's something else that I don't know :)

**Comment by SharakPL** [comment]
IMO it would be better to just use the zoom directly on the main image container without opening it on full screen backdrop on mobile. Only on desktops there should be option to open big size image besides zooming.

**Comment by Hlavtox** [comment]
@SharakPL Try it on my store, you can zoom properly by gestures, swipe to change images, double tap to zoom in and out, it shows image description on the bottom etc.

**Comment by SharakPL** [comment]
I did, but it still loads new full screen carousel which on mobile doesn't really change anything apart from disabling access to the thumbs below main image.



IMO it should provide this zoom-in feature on doubletap directly in the main image container and full screen feature should be restricted to desktop size only.

**Comment by SharakPL** [comment]
[Here's what I mean](https://fancyapps.com/docs/ui/panzoom/#quick-demo) by zooming inside image container.

**Review by kpodemski — APPROVED** [review]
I think we should accept this solution. It's cleaner than using a third-party library.

WDYT @Hlavtox? Maybe we could have it configurable or something.

**Comment by Hlavtox** [comment]
@kpodemski I wanted to do a real gallery plugin with zooming, swipe etc. :(

**Comment by NeOMakinG** [comment]
Tbh I like it, and you could create a tutorial on the storybook to explain how you would use a third party library to achieve a more powerful feature @Hlavtox

File: src/scss/custom/components/_slider.scss [hunk 1]
```diff
@@ -10,9 +10,9 @@ $component-name: carousel;
     justify-content: center;
     height: 100%;
 
-    &.active {
-      display: flex;
-    }
+    // &.active {
+    //   display: flex;
+    // }
```
Comment (kpodemski) [inline src/scss/custom/components/_slider.scss:15]:
```suggestion
```

**Review by kpodemski — APPROVED** [review]
I'm merging it. I think we all agree that it is an improvement, and I don't want @JBLach to be stale for the next half a year, especially since no one is motivated to propose a different version.

**Comment by SharakPL** [comment]
This would be nice to implement in the future for product gallery https://codepen.io/argyleink/pen/dyLNgpX 🤩
Doesn't animate on Firefox yet

**Comment by dennispw** [comment]
> This would be nice to implement in the future for product gallery https://codepen.io/argyleink/pen/dyLNgpX 🤩 But better without generating shifts in the thumbnails.
> 
> Doesn't animate on Firefox yet

As long as it is completely opt-in instead of opt-out.

**Pull request merged**